rocky/rocky/views/organization_member_add.py

```
@@ -54,3 +69,89 @@ def form_valid(self, form):
+
+                try:
+                    user, user_created = User.objects.get_or_create(email=email, defaults={"full_name": full_name})
+                except Exception:
+                    logger.exception("Failed creating user")
+                    continue
+                if user_created:
+                    form = PasswordResetForm({"email": email})
+
+                    if not form.is_valid():
+                        logger.warning("Email not valid: %s", email)
+                        continue
+
+                    form.save(
+                        email_template_name="registration_email.html",
+                        subject_template_name="registration_subject.txt",
+                        extra_email_context={"organization": self.organization.name},
+                        use_https=self.request.is_secure(),
+                        request=self.request,
+                    )
+
+                try:
+                    member, member_created = OrganizationMember.objects.get_or_create(
+                        user=user,
+                        defaults={"organization": self.organization},
+                    )
+                except Exception:
+                    logger.exception("Failed creating organization member")
+                    if user_created:
+                        user.delete()
+
+                    continue
+
+                if member_created:
+                    member.organization = self.organization
+                    member.trusted_clearance_level = trusted_clearance_level
+                    member.acknowledged_clearance_level = acknowledged_clearance_level
+
+                try:
+                    member.groups.add(Group.objects.get(name=account_type))
+                except ObjectDoesNotExist:
+                    logger.exception("Failed adding organization member to group")
+                    continue
+
+            messages.add_message(self.request, messages.SUCCESS, _("Successfully added csv."))
```

The message is a bit misleading since errors are silenced/ not visible to the end user. Perhaps something like "CSV processed" suits better?

---

Yess